### PR TITLE
style: sidebar ui enhancement

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Sidebar/Sidebar.res [hunk 1]
```diff
@@ -407,13 +410,19 @@ module SidebarNestedSection = {
     <RenderIf condition={!areAllSubLevelsHidden}>
       <NestedSectionItem
         section
-        isSectionExpanded
+        isSectionExpanded={isSectionAutoCollapseEnabled
```
Comment (PritishBudhiraja) [inline src/screens/Sidebar/Sidebar.res:413]: Put this condition out for better readabiluty and use If else for the same

File: src/screens/Sidebar/Sidebar.res [hunk 2]
```diff
@@ -407,13 +410,19 @@ module SidebarNestedSection = {
...
         isAnySubItemSelected
         textColor
         cursor
-        toggleSectionExpansion
+        toggleSectionExpansion={isSectionAutoCollapseEnabled
+          ? _ => setOpenItem(prev => {prev == section.name ? "" : section.name})
+          : toggleSectionExpansion}
         expandedTextColor
-        isElementShown
+        isElementShown={isSectionAutoCollapseEnabled
```
Comment (PritishBudhiraja) [inline src/screens/Sidebar/Sidebar.res:423]: Same

File: src/screens/Sidebar/Sidebar.res [hunk 3]
```diff
@@ -536,6 +545,8 @@ let make = (
     )
   }
 
+  let (openItem, setOpenItem) = React.useState(_ => "")
```
Comment (PritishBudhiraja) [inline src/screens/Sidebar/Sidebar.res:548]: declare state in the starting of the component.

File: src/screens/Sidebar/Sidebar.res (hunk 1 above)
Comment (gitanjli525) [reply src/screens/Sidebar/Sidebar.res]: done

File: src/screens/Sidebar/Sidebar.res [hunk 4]
```diff
@@ -407,13 +410,19 @@ module SidebarNestedSection = {
...
         section
-        isSectionExpanded
+        isSectionExpanded={isSectionAutoCollapseEnabled
+          ? openItem === section.name || isAnySubItemSelected
+          : isSectionExpanded}
         isAnySubItemSelected
         textColor
         cursor
-        toggleSectionExpansion
+        toggleSectionExpansion={isSectionAutoCollapseEnabled
```
[the remark this answers is not in the document]
Comment (gitanjli525) [reply src/screens/Sidebar/Sidebar.res]: done

File: src/screens/Sidebar/Sidebar.res (hunk 2 above)
Comment (gitanjli525) [reply src/screens/Sidebar/Sidebar.res]: done

File: src/screens/Sidebar/Sidebar.res (hunk 3 above)
Comment (gitanjli525) [reply src/screens/Sidebar/Sidebar.res]: okay, sure

File: src/screens/Sidebar/Sidebar.res [hunk 5]
```diff
@@ -404,16 +407,35 @@ module SidebarNestedSection = {
       | SubLevelLink({access}) => access === NoAccess
       }
     })
+
+    let isSectionExpanded_ = if isSectionAutoCollapseEnabled {
```
Comment (PritishBudhiraja) [inline src/screens/Sidebar/Sidebar.res:411]: No need for underscore

File: src/screens/Sidebar/Sidebar.res [hunk 6]
```diff
@@ -404,16 +407,35 @@ module SidebarNestedSection = {
...
+      isSectionExpanded
+    }
+
+    let toggleSectionExpansion_ = if isSectionAutoCollapseEnabled {
+      _ => setOpenItem(prev => {prev == section.name ? "" : section.name})
+    } else {
+      toggleSectionExpansion
+    }
+
+    let isElementShown_ = if isSectionAutoCollapseEnabled {
```
Comment (PritishBudhiraja) [inline src/screens/Sidebar/Sidebar.res:423]: No need for underscore

File: src/screens/Sidebar/Sidebar.res [hunk 7]
```diff
@@ -404,16 +407,35 @@ module SidebarNestedSection = {
...
       }
     })
+
+    let isSectionExpanded_ = if isSectionAutoCollapseEnabled {
+      openItem === section.name || isAnySubItemSelected
+    } else {
+      isSectionExpanded
+    }
+
+    let toggleSectionExpansion_ = if isSectionAutoCollapseEnabled {
```
Comment (PritishBudhiraja) [inline src/screens/Sidebar/Sidebar.res:417]: No need for underscore

**Review by PritishBudhiraja — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**